### ISSUE TITLE
Establish SVG 2 test suite

### ISSUE THREAD

**Issue by nikosandronikos** [w3c/svgwg#269]
We need to get the SVG 2 test suite up and running.

There are some plans in place, but most in the working group don't know what they are or how they are supposed to submit tests.

This meta bug will track the initial creation of the test suite to the point where all working group members can start submitting tests.

**Comment by nikosandronikos** [comment]
I've created a wiki page where we can capture decisions
https://github.com/w3c/svgwg/wiki/Testing

The current plans are:
* host on https://github.com/w3c/web-platform-tests/tree/master/svg
* Sort tests in directories by chapter, and then by whether they are a reftest or a script test

Unknowns / to dos:
* Do we want to use Shepherd? See #270.
* Define a template for SVG reftests
* Define a template for SVG script tests
* Work out who should accept PRs and get them access

**Comment by BigBadaboom** [comment]
I have about 180 test files that I wrote while working on my library.  I'm
happy to contribute anything I have, although they will probably need to be
massaged a bit.  There is probably some overlap with the test suite,
however many of the tests were written in order to exercise features that
weren't well covered by the test suite.

Paul

**Comment by Tavmjong** [comment]
We defined templates a few years ago: https://www.w3.org/Graphics/SVG/WG/wiki/SVG2/Testing_Requirements

**Comment by gsnedders** [comment]
cc/ @jgraham @ms2ger

**Comment by gsnedders** [comment]
> Sort tests in directories by chapter, and then by whether they are a reftest or a script test

Is there any reason to do this? (Also, there'll have to be a fair number of manual tests, as animation tests mostly cannot be automated.)

> Define a template for SVG reftests
> Define a template for SVG script tests

I'll point out that I think you're likely to come up against some opposition if you want to require specific templates in web-platform-tests. 

> Work out who should accept PRs and get them access

**Comment by gsnedders** [comment]
I'll also point out that taking tests from Gecko/WebKit/Blink may well be a good place to get started (anything in mozilla-central post 23rd September 2014 or by a Mozilla employee is CC0) and WebKit/Blink should't be too hard to get them to contribute tests.

**Comment by AmeliaBR** [comment]
I'm really hopeful that we can port any and all existing testing suites to get things up and running for a start.

However, we'll also need a specific process of reviewing all these to make sure they match up with specific SVG 2 text (some of which has changed re edge cases), and that all edge cases are covered.  This was a big problem with the SVG 1.1 test suite: it wasn't very well connected to the spec.

Given that reviews are going to be required, I don't see requiring a template as too big an obstacle.  There are a number of SVG-specific metadata we'll need, particularly re which processing modes the results are for (static, animated, scripted, with external files allowed or not), and whether it works for SVG standalone, SVG as inline in XML, and/or SVG inline in HTML.

@gsnedders What's your concerns about organizing tests by chapter?  We're going to be looking at 1000s of tests, it would be nice to have some sort of structure to them.

**Comment by Tavmjong** [comment]
We need to make sure non-browsers can render and report test results.

**Comment by nikosandronikos** [comment]
@AmeliaBR @gsnedders I suspect the issue was organising within the chapter into reftest, script test, etc directories.
As @Tavmjong alluded to, the reason we wanted to do this was to help non browser implementations such as Inkscape sort the tests they want to comply with (reftests) from the others that they don't support (script tests).

**Comment by Ms2ger** [comment]
> Given that reviews are going to be required, I don't see requiring a template as too big an obstacle.

I strongly disagree: there's a significant difference between "check that this matches the spec and potentially add some cases" and "rewrite every single test from scratch".

> the reason we wanted to do this was to help non browser implementations such as Inkscape sort the tests they want to comply with (reftests) from the others that they don't support (script tests).

There's no need to do that by separating the files into distinct directories; we have scripts that will output a manifest of tests sorted by type already.

**Comment by Ms2ger** [comment]
One general point that bears repeating: *every* roadblock you put up, however excellent a reason you have to put it up, will reduce the number of tests that is submitted.

Also, every requirement for tests you have (besides being correct), should be checked by the lint tool. (I will merge correct tests without prejudice if they have no lint failures.)

**Comment by nikosandronikos** [comment]
FYI, I've updated https://github.com/w3c/svgwg/wiki/Testing. And I think this is pretty close to the final set of recommendations for producing the SVG 2 test suite.

I've also had a go at converting some SVG 1.1 tests to reftests:
https://github.com/w3c/web-platform-tests/pull/4015

**Comment by boggydigital** [comment]
Removing agenda+ item per discussion during the call, keeping item active

**Comment by boggydigital** [comment]
This is a work item required to publish SVG 2.0 specification, not a comment on anything in the specification and is not blocking updated CR publication. 

We'll keep this active and target SVG 2.0 Recommendation milestone
